src/crimson/osd/osd_operations/internal_client_request.cc

```diff
@@ -0,0 +1,134 @@
+      logger().debug("{}: in repeat", *this);
+      return interruptor::with_interruption([this]() mutable {
+        return with_blocking_future_interruptible<IOInterruptCondition>(
+          handle.enter(pp().wait_for_active)
+        ).then_interruptible([this] {
+          return with_blocking_future_interruptible<IOInterruptCondition>(
+            pg->wait_for_active_blocker.wait());
+        }).then_interruptible([this] {
+          return with_blocking_future_interruptible<IOInterruptCondition>(
+            handle.enter(pp().recover_missing)
+          ).then_interruptible([this] {
+            return do_recover_missing(pg, {});
+          }).then_interruptible([this] {
+            return with_blocking_future_interruptible<IOInterruptCondition>(
+              handle.enter(pp().get_obc)
+            ).then_interruptible([this] () -> PG::load_obc_iertr::future<> {
+              logger().debug("{}: getting obc lock", *this);
+              return fabricate_osd_ops().then([this] (auto&& osd_ops) {
+                logger().debug("InternalClientRequest: got {} OSDOps to execute",
+                               std::size(osd_ops));
+                [[maybe_unused]] const int ret = op_info.set_from_op(
+                  std::as_const(osd_ops), pg->get_pgid().pgid, *pg->get_osdmap());
+                assert(ret == 0);
+                return pg->with_locked_obc(get_target_oid(), op_info,
+                  [osd_ops=std::move(osd_ops), this](auto obc) {
+                  return with_blocking_future_interruptible<IOInterruptCondition>(
+                    handle.enter(pp().process)
+                  ).then_interruptible(
+                    [obc=std::move(obc), osd_ops=std::move(osd_ops), this] {
+                    return pg->do_osd_ops(
+                      std::move(obc),
+                      std::move(osd_ops),
+                      std::as_const(op_info),
+                      get_do_osd_ops_params(),
+                      [this] {
+                        return PG::do_osd_ops_iertr::now();
+                      },
+                      [this] (const std::error_code& e) {
+                        return PG::do_osd_ops_iertr::now();
+                      }
+                    ).safe_then_interruptible(
+                      [this]() -> interruptible_future<> {
+                        return interruptor::now();
+                      }, crimson::ct_error::eagain::handle([this]() mutable {
+                        return interruptor::now();
+                      })
+                    );
+                  });
+                });
+              });
+            }).handle_error_interruptible(PG::load_obc_ertr::all_same_way([] {
+              return seastar::now();
+            })).then_interruptible([] {
+              return seastar::stop_iteration::yes;
+            });
+          });
+        });
+      }, [this](std::exception_ptr eptr) mutable {
+        if (*eptr.__cxa_exception_type() ==
```

Comment:
nit, could you extract this part into a helper method, something like 
```c++
static bool should_abort(std::exception_ptr eptr) {
  // 
}
/// ...
if (should_abort(std::move(eptr)) {
  return seastar::stop_iteration::yes;
} else {
  seastar::stop_iteration::no;
}
```
```

Reply:
yeah, it makes sense to put this low-level, `__cxa_exception_type`-aware check into a dedicated, reusable function. Will send a dedicated commit for that.

Reply:
Dissected into `CommonClientRequest`.